Check common locations for yapf and persist to settings

jason-kane opened the pull request:
Simple approach -- check common yapf install locations.  What I don't know is what the common locations actually are.  It might be better to just bite the bullet and search $PATH but ugg.  Maybe call `which` (though I don't think `which` exists in windows).

PR to collect other paths yapf will land in before pushing a release.

catch22 commented on PyYapf.py:
```diff
@@ -190,6 +178,40 @@ def __exit__(self, type, value, traceback):
+        """Find the yapf executable."""
+        # default to what is in the settings file
+        cmd = self.get_setting("yapf_command")
+        if not cmd:
```
It's perhaps strange that we still show an error message in this case. I would actually suggest to remove the default `yapf_command' and make this case (when `cmd` is empty) the only case that the "search common possibilities" logic is invoked.

jason-kane replied:
It is a weird edge case; hitting it means the user edited the settings file and removed the default yapf_command (or otherwise made it evaluate as false).

reg. triggering on '' (and making that the default) -- perfect.  That takes care of the new user issue (which is the real issue) without adding magic for the "normal" case.  Maybe tweak the fail string to hint to try "" as yapf_command to trigger a search.

catch22 commented:
See my inline comment for an alternative suggestion that would never overwrite the user's choice of `yapf_command`.

Brainstorming a bit further: Instead of adding a list of hardcoded paths (which might be a bit painful to maintain in the future) how about we just default to `yapf` as the `yapf_command` and search the user's PATH for it (whenever the configuration file gets reloaded), e.g. using https://github.com/trentm/which, which is a nice and MIT-licensed Python replacement for `which`. We could in addition pop up the README file when the package is installed and state that by default we expect the `yapf` binary to be found in the PATH.

jason-kane commented:
dependencies in a sublime package are awkward -- it isn't like we get pip installed and have a setup.py / requirements.txt etc... I suspect there is a short list that captures 95% but could easily be wrong.

Great suggestion reg. empty == search.  Messing with a users config (however well intentioned) is pretty rude, only doing it once at install time isn't nearly as bad.  I changed the behavior and updated the PR.

catch22 commented:
I was thinking of integrating `which.py` (it's MIT licensed).

jason-kane commented:
yank it into the repo?  given that it hasn't been touched in 9 years I guess we don't have to worry about keeping up with bug fixes. :)

catch22 commented:
Sorry, I just remembered that I was actually using `shutil.which` in the past! No dependency and maintained by Guido & friends ;-) It's Python 3.3+ only.

jason-kane commented:
https://github.com/minrk/backports.shutil_which doesn't look like a horrible option
